tests/integration/test_clip.py

```diff
@@ -0,0 +1,180 @@
+    # res_np is not match res_num
+    # in Numpy, if a_min or a_max is float, all data are marked as float,
+    # while in cunumeric, all datas are int.
+    # for example, amin = 5
+    # array = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    # res_np = array([5., 5., 5., 5., 5., 5., 6., 7., 8., 8.5])
+    # res_num = array([5, 5, 5, 5, 5, 5, 6, 7, 8, 8])
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.xfail
+def test_amin_complex():
+    array = np.arange(0, 10)
+    amin = 5 + 5j
+    res_np = np.clip(array, a_min=amin, a_max=8.5)
+    #  res_np = array([5. +5.j, 5. +5.j, 5. +5.j, 5. +5.j, 5. +5.j,
+    #   5. +5.j, 6. +0.j, 7. +0.j, 8. +0.j, 8.5+0.j])
+    res_num = num.clip(array, a_min=amin, a_max=8.5)
+    # cunumeric raises:
+    # TypeError: int() argument must be a string, a bytes-like object
+    # or a real number, not 'complex'
+    assert np.array_equal(res_np, res_num)
+
+
+def test_value_list():
+    array = np.arange(0, 5)
+    amin = [2, 3, 4, 5, 1]
+    amax = 8
+    res_np = np.clip(array, a_min=amin, a_max=amax)
+    res_num = num.clip(array, a_min=amin, a_max=amax)
+    assert np.array_equal(res_np, res_num)
+
+
+def test_value_list_negative():
+    array = np.arange(0, 10)
+    amin = [2, 3, 4, 5, 1]
+    amax = 8
+    expected_exc = ValueError
+    with pytest.raises(expected_exc):
+        np.clip(array, a_min=amin, a_max=amax)
+        # ValueError: operands could not be broadcast together
+    with pytest.raises(expected_exc):
+        num.clip(array, a_min=amin, a_max=amax)
+
+
+def test_out_negative():
+    array = np.arange(0, 5)
+    out_a = np.arange(0, 3)
+    amin = [2, 3, 4, 5, 1]
+    amax = 8
+    expected_exc = ValueError
+    with pytest.raises(expected_exc):
+        np.clip(array, a_min=amin, a_max=amax, out=out_a)
+    with pytest.raises(expected_exc):
+        num.clip(array, a_min=amin, a_max=amax, out=out_a)
+
+
+def test_out_negative_ndim():
+    array = [[2, 3, 4], [3, 4, 5], [6, 6, 12]]
+    np_arr = np.array(array)
+    num_arr = num.array(array)
+    out_a = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+    out_np = np.array(out_a)
+    out_num = num.array(out_a)
+    amin = 3
+    amax = 8
+    np.clip(np_arr, a_min=amin, a_max=amax, out=out_np)
+    num.clip(num_arr, a_min=amin, a_max=amax, out=out_num)
+    assert np.array_equal(out_np, out_num)
+
+
+@pytest.mark.parametrize("ndim", range(1, LEGATE_MAX_DIM + 1))
+def test_basic(ndim):
+    shape = (5,) * ndim
+    np_arr = mk_seq_array(np, shape)
+    num_arr = mk_seq_array(num, shape)
+
+    amin = int(np.prod(shape) / 2)
+    amax = np.prod(shape) - 1
+    res_np = np.clip(np_arr, amin, amax)
+    res_num = num.clip(num_arr, amin, amax)
+    assert np.array_equal(res_num, res_np)
+
+
+@pytest.mark.parametrize("ndim", range(1, LEGATE_MAX_DIM + 1))
+def test_out(ndim):
+    shape = (5,) * ndim
+    np_arr = mk_seq_array(np, shape)
+    num_arr = mk_seq_array(num, shape)
+
+    out_np = np.zeros(shape)
+    out_num = num.zeros(shape)
```

Comment:
no need to initialize, `empty` would suffice

Reply:
fix